### PR TITLE
pl022 updates/fixes

### PR DESCRIPTION
**Pull request by vchong**

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by vchong** [comment]
Rebased, tag added and fixed IBART warning about indentations.
Hopefully ok to merge.
Thanks.

**Comment by jbech-linaro** [comment]
@vchong , forgot to add Jens A-B in `pl022_spi: simplify receive of remaining data`?

**Comment by vchong** [comment]
> @vchong , forgot to add Jens A-B in pl022_spi: simplify receive of remaining data?

Oops, sorry! Fixed!

**Comment by vchong** [comment]
Sorry again. I added a 3rd (minor) patch. Please review it as well. Thanks.

File: core/drivers/pl061_gpio.c [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #include <assert.h>
 #include <drivers/pl061_gpio.h>
 #include <io.h>
+#include <mm/core_memprot.h>
```
Comment (jenswi-linaro) [inline core/drivers/pl061_gpio.c]: Why are these added? For the `KEEP_PAGER()` macro you need to add  `<keep.h>` though.

**Comment by jenswi-linaro** [comment]
Please re-format the last commit message

File: core/drivers/pl061_gpio.c (hunk 1 above)
Comment (vchong) [reply core/drivers/pl061_gpio.c]: Fixed

**Comment by vchong** [comment]
Commit message re-formatted. All comments addressed.

**Comment by jenswi-linaro** [comment]
For the commit "pl022, pl061: add missing pager constraint on _ops struct ":
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by vchong** [comment]
Tag added for 3rd patch and re-based again. Please merge if ok. Thanks!

File: core/drivers/pl022_spi.c [hunk 2]
```diff
@@ -166,11 +166,9 @@ static enum spi_result pl022_txrx8(struct spi_chip *chip, uint8_t *wdat,
 	/* Capture remaining rdat not read above */
 	if (rdat) {
 		while ((j < num_pkts) &&
-			(io_read8(pd->base + SSPSR) & SSPSR_BSY))
-			if (io_read8(pd->base + SSPSR) & SSPSR_RNE) {
-				/* rx 1 packet */
-				rdat[j++] = io_read8(pd->base + SSPDR);
-			}
+			(io_read8(pd->base + SSPSR) & SSPSR_RNE))
```
Comment (etienne-lms) [inline core/drivers/pl022_spi.c]: may I suggest to play with indentation and replace last tab above with 7 space chars and add `{}` since block is made of a comment and an instruction?



```
		while ((j < num_pkts) &&
		       (io_read8(pd->base + SSPSR) & SSPSR_RNE)) {
			/* rx 1 packet */
			rdat[j++] = io_read8(pd->base + SSPDR);
		}
```

Same comment for line 213 below.

File: core/drivers/pl022_spi.c [hunk 3]
```diff
@@ -212,11 +210,9 @@ static enum spi_result pl022_txrx16(struct spi_chip *chip, uint16_t *wdat,
...
+			/* rx 1 packet */
+			rdat[j++] = io_read16(pd->base + SSPDR);
```
Comment (etienne-lms) [inline core/drivers/pl022_spi.c]: Nitpicking: propose to move the fix `io_read8()`->`io_read16()` from commit "pl022_spi: simplify receive of remaining data" to commit "pl022_spi: fix non-trivial typo".

File: core/drivers/pl022_spi.c (hunk 2 above)
Comment (vchong) [reply core/drivers/pl022_spi.c]: done

File: core/drivers/pl022_spi.c (hunk 3 above)
Comment (vchong) [reply core/drivers/pl022_spi.c]: Hrm, that was the way it was done originally. Must have messed up the rebase. Anyway, fixed.

**Comment by vchong** [comment]
@etienne-lms Thanks! All comments addressed and tags added.
